lib/twingly/http.rb

```diff
@@ -81,12 +93,20 @@ def initialize_defaults
       end
 
       # rubocop:disable Metrics/MethodLength
+      # rubocop:disable Metrics/AbcSize
+      # rubocop:disable Metrics/CyclomaticComplexity
```

Comment:
Managed to get rid of them by using the following instead of a `case` statement. Don't really like it, but at least we got rid of the RuboCop complaints, and we didn't have to make any changes to the get/post methods above.

```diff
diff --git a/lib/twingly/http.rb b/lib/twingly/http.rb
index 90dffdb..5735541 100644
--- a/lib/twingly/http.rb
+++ b/lib/twingly/http.rb
@@ -92,22 +92,8 @@ module Twingly
         @max_url_size_bytes     = DEFAULT_MAX_URL_SIZE_BYTES
       end
 
-      # rubocop:disable Metrics/MethodLength
-      # rubocop:disable Metrics/AbcSize
-      # rubocop:disable Metrics/CyclomaticComplexity
       def http_response_for(method, **args)
-        response = case method
-                   when :get
-                     http_get_response(**args)
-                   when :post
-                     http_post_response(**args)
-                   when :put
-                     http_put_response(**args)
-                   when :patch
-                     http_patch_response(**args)
-                   when :delete
-                     http_delete_response(**args)
-                   end
+        response = method("http_#{method}_response").call(**args)
 
         Response.new(headers: response.headers.to_h,
                      status: response.status,
@@ -119,7 +105,6 @@ module Twingly
       rescue FaradayMiddleware::RedirectLimitReached => error
         raise RedirectLimitReachedError, error.message
       end
-      # rubocop:enable all
 
       def http_get_response(url:, params:, headers:)
         binary_url = url.dup.force_encoding(Encoding::BINARY)
```

Reply:
not that much better but possibly you could do the same with `send` or `public_send`, haven't tested but I mean something along the lines of `public_send(method, **args)`

Reply:
> Would be nice if we could get rid of these, but we can skip that for now, as it involves making changes to the `#get` and `#post` methods too 🙈

This was my initial thought as well, but as you say the suggestions here can be implemented pretty discretely. 

Both `#method().call` and `#send` works. I think the former is a bit more idiomatic but I'm fine with both! Do you have any preferences? :)

Reply:
How did you reach the conclusion the former is more idiomatic? I believe I have seen more `send` than `method(...).call` 😄 

I was curious on the practical difference, seems pretty similar:

```ruby
[1] pry(main)> class SomeClass end;
[2] pry(main)> SomeClass.new.method(:apa).call("1")
NameError: undefined method `apa' for class `SomeClass'
from (pry):2:in `method'
[3] pry(main)> SomeClass.new.public_send(:apa, 1)
NoMethodError: undefined method `apa' for #<SomeClass:0x00007fed38aa1f68>
from (pry):3:in `public_send'
```

Reply:
`NoMethodError` probably slightly better than `NameError`? 🤷

Reply:
> How did you reach the conclusion the former is more idiomatic? I believe I have seen more send than method(...).call 😄

Me too! Sorry, didn't mean idiomatic, only that it's super clear what it does :)

> NoMethodError probably slightly better than NameError? 🤷

Agreed!

Reply:
Changed to `#send` in afd9405 👍

Reply:
I can just add that I didn't have a particular reason for using `method(...).call` over `send(...)`. It was just the first thing that popped into my head for some reason 😃